src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -342,14 +363,19 @@ public new DocumentState UpdateText(TextLoader loader, PreservationMode mode)
                 ? CreateStrongText(loader, this.Id, this.solutionServices, reportInvalidDataException: true)
                 : CreateRecoverableText(loader, this.Id, this.solutionServices, reportInvalidDataException: true);
 
-            var newTreeSource = CreateLazyFullyParsedTree(
-                newTextSource,
-                this.Id.ProjectId,
-                GetSyntaxTreeFilePath(this.info),
-                _options,
-                _languageServices,
-                this.solutionServices,
-                mode);
+            // Only create the ValueSource for creating the SyntaxTree if this is a Document that
+            // supports SyntaxTrees.  There's no point in creating the async lazy and holding onto
+            // this data otherwise.
+            var newTreeSource = !this.SupportsSyntaxTree
```

Comment:
small thing but can these two "?:" use same order of condition, either this.SupportSyntax .. or !this.SupportSyntax...